src/Compiler/Driver/CompilerConfig.fs

```diff
@@ -877,6 +878,14 @@ type TcConfigBuilder =
             compilationMode = TcGlobals.CompilationMode.Unset
         }
 
+    member tcConfigB.SetLangVersion v =
```

Comment:
Adding a mutator for langVersion is fine but mostly unnecessary, but please don't let the langversion mutator set other values that is not the paradigm we use.
TcConfigBuilder is a mutable record and so modifying values is kind of what it's about.

Reply:
When `tcConfigB.diagnosticsOptions` is set to `FSharpDiagnosticOptions.Default`, the langversion is not yet known. The langversion comes only much later with `ParseCompilerOpions`.
There are two alternatives that I considered but didn't like.
The first was to add to `tcConfigB` an updated `FSharpDiagnosticOptions` (updated with the langversion) in all the places where `ParseCompilerOptions` is called (in fsc.fs, fsi.fs, BackgroundCompiler.fs, TransparentCompiler.fs and FSharpCheckerResults.fs).
The second was to thread the whole `tcConfig` through all the function calls to the places where I need only the diagnosticsOptions.
But let me check if I find still another way.